fs/btrfs/zlib.c

```diff
@@ -36,7 +36,19 @@ static void zlib_free_workspace(struct list_head *ws)
 	kfree(workspace);
 }
 
-static struct list_head *zlib_alloc_workspace(void)
+static int zlib_set_level(struct list_head *ws, unsigned int level)
+{
+	struct workspace *workspace = list_entry(ws, struct workspace, list);
+
+	if (level > 9)
+		level = 9;
+
+	workspace->level = level > 0 ? level : 3;
```

Comment:
Should there be `BTRFS_ZLIB_DEFAULT_LEVEL` and `BTRFS_ZLIB_MAX_LEVEL` constants to avoid magic numbers?